test/e2e/fixtures/then.go

```diff
@@ -134,6 +134,14 @@ func (t *Then) ExpectWorkflowList(listOptions metav1.ListOptions, block func(t *
 	return t
 }
 
+func (t *Then) ExpectWorkflowListRetryOnError(listOptions metav1.ListOptions, block func(t *testing.T, client v1alpha1.WorkflowInterface, listOptions metav1.ListOptions)) *Then {
```

Comment:
why does other func not work? I don't understand this based on its name, the implementation does not seem to match the name.

Reply:
This function is needed to void the wait, I think we can reuse this function/pattern instead of the `wait` that is used in other places as well. WDYT @alexec ?